tinygrad/codegen/linearizer.py

```diff
@@ -433,8 +443,7 @@ def ast_parse(self, x:LazyOp, accs: Dict[LazyOp, List[UOp]], offs:Optional[List[
       return [self.uops.add(UOps.BITCAST if x.op is UnaryOps.BITCAST else UOps.CAST,
                             self.get_base_dtype(x.arg), (u,)) for u in self.ast_parse(x.src[0], accs, offs, loaded_buffers)]
     if x.op in ReduceOps and reduce_acc is None:
-      assert offs is None, "not available if we aren't doing reduce"
-      return accs[x]
+      return [accs[x][i] for i in offs] if offs else accs[x]
```

Comment:
nit: `offs is not None`

Reply:
working on the added tests and `UNROLL` is (I think the only) case where offs and accs don't match:

ex you get a kernel like this:
```
__kernel void r_2_3(__global float* data0, const __global float* data1) {
  int gidx0 = get_group_id(0); /* 2 */
  int alu0 = (gidx0*3);
  float val0 = data1[alu0];
  float val1 = data1[alu0+1];
  float val2 = data1[alu0+2];
  float alu1 = (val2+val1+val0);
  data0[gidx0] = ((val2-alu1)+(val1-alu1)+(val0-alu1));
}
```
where there's one acc, `alu1`, but three offs, resulting in: `(val2-alu1)`, `(val1-alu1)`, and `(val0-alu1)`

to `ast_parse` this looks like:
```
accs[x] =  UOps.PHI            : dtypes.float              [<UOps.DEFINE_ACC: 5>, <UOps.ALU: 13>] None
offs = [0, 0, 0]
```

Reply:
but this only happens when there's multiple reduces because it reduces the unrolled shape down to one value then has to use it in the pre-unrolled shape for farther arithmetic